automapper/extensions/default.py

```diff
@@ -15,6 +15,7 @@ def __init_method_classifier__(target_cls: Type[T]) -> bool:
         and isinstance(
             getattr(getattr(target_cls, "__init__"), "__annotations__"), dict
         )
+        and getattr(getattr(target_cls, "__init__"), "__annotations__")
```

Comment:
Could you explain please this line?
The line gets the attribute and checks if it's an instance of `dict`.
So the new line gets an attribute and checks if it's not `None`? I think it is included already in the line above, no?

Reply:
Thanks for the fast reply. It check for the corner-case where it is not `None`, but the empty dict `{}`. This also evaluates to `False` as a bool and this is not covered by the previous lines. And this is precisely what an SQLAlchemy instrumented class has in __annotations__.

It took me a while to spot this, as the unit test did not consistently fail. You loop through `reversed(...items())` when trying to locate a matching spec. But the order in the iteration is not always the same, I believe. As far as I can tell, the `.items()` order should be the insertion order. But probably the insertions are not consistent when looping over the *.py files in `extensions/`? I didn't investigate further.